sql: convert NULL to false

madelynnblue opened the pull request:
Matches behavior of postgres, which returns 2 for:

select case when null then 1 else 2 end;

Where previously we would exit with:

query error: cannot convert NULL to bool

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/3544)

<!-- Reviewable:end -->

tamird commented on sql/parser/datum.go:
```diff
@@ -88,6 +88,9 @@ func getBool(d Datum) (DBool, error) {
 	if v, ok := d.(DBool); ok {
 		return v, nil
 	}
+	if _, ok := d.(dNull); ok {
```
compare directly to DNull and add a simple test

petermattis commented:
This PR needs an associated test.

madelynnblue committed:
sql: convert NULL to false

Matches behavior of postgres, which returns 2 for:

select case when null then 1 else 2 end;

Where previously we would exit with:

query error: cannot convert NULL to bool

mjibson commented:
Test added.

---

Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/parser/datum.go, line 91 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3544#-K6fwF2CjD3fSiPw8z_f-r1-91:-K6jgmh03SK5mFZvYwDn:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/516137e957b1b8851998f21410dc0a75f75535ce/sql/parser/datum.go#L91)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3544#-:-K6jglRt5Ye9mk5qI32X:-372168649)*
<!-- Sent from Reviewable.io -->

petermattis commented:
LGTM
